oauthenticator/github.py

```diff
@@ -215,8 +229,68 @@ async def authenticate(self, handler, data=None):
                     auth_state['github_user']['email'] = val['email']
                     break
 
+        if self.populate_teams_in_auth_state:
+            if 'read:org' not in self.scope:
+                # This means the 'read:org' scope was not set, and we can't fetch teams
+                self.log.error(
+                    'read:org scope is required for populate_teams_in_auth_state functionality to work'
+                )
+            else:
+                # Number of teams to request per page
+                per_page = 100
+
+                #  https://docs.github.com/en/rest/reference/teams#list-teams-for-the-authenticated-user
+                url = self.github_api + f"/user/teams?per_page={per_page}"
+
+                auth_state['teams'] = await self._paginated_fetch(url, access_token)
+
         return userdict
 
+    async def _paginated_fetch(self, api_url, access_token):
+        """
+        Fetch all items via a paginated GitHub API call
+
+        Makes a request to api_url, and if pagination information is returned,
+        keep paginating until all the items are retrieved.
+        """
+        url = api_url
+        content = []
+        while True:
+            req = HTTPRequest(
+                url,
+                method="GET",
+                headers=_api_headers(access_token),
+                validate_cert=self.validate_server_cert,
+            )
+            resp = await self.fetch(req, "fetching user teams", parse_json=False)
+
+            resp_json = json.loads(resp.body.decode())
```

Comment:
In case GitHub is down or we get a slow-down 429 response or similar it would be good to error with a message about status code instead of failing to parse the json.

```suggestion
            resp.raise_for_status()
            resp_json = json.loads(resp.body.decode())
```